mache/spack/chicoma-cpu_gnu_mpich.yaml

```diff
@@ -91,67 +92,55 @@ spack:
       buildable: false
     gcc:
       externals:
-      - spec: gcc@12.2.0
+      - spec: gcc@12.3
         modules:
-        - PrgEnv-gnu/8.4.0
-        - gcc/12.2.0
-        - craype-accel-host
-        - craype
+        - PrgEnv-gnu/8.5.0
+        - gcc-native/12.3
         - libfabric/1.15.2.0
+        - cray-libsci/23.12.5
+        - craype/2.7.30
+        - craype-accel-host
+        - craype-x86-rome
       buildable: false
     cray-mpich:
       externals:
-      - spec: cray-mpich@8.1.26
-        prefix: /opt/cray/pe/mpich/8.1.26/ofi/gnu/9.1
+      - spec: cray-mpich@8.1.28
         modules:
-        - libfabric/1.15.2.0
-        - cray-mpich/8.1.26
+        - cray-mpich/8.1.28
       buildable: false
-    libfabric:
-      externals:
-      - spec: libfabric@1.15.2.0
-        prefix: /opt/cray/libfabric/1.15.2.0
-        modules:
-        - libfabric/1.15.2.0
-      buildable: false
-{% if e3sm_lapack %}
     cray-libsci:
       externals:
-      - spec: cray-libsci@23.05.1.4
-        prefix: /opt/cray/pe/libsci/23.05.1.4/GNU/9.1/x86_64
+      - spec: cray-libsci@23.12.5
         modules:
-        - cray-libsci/23.05.1.4
+        - cray-libsci/23.12.5
       buildable: false
-{% endif %}
 {% if e3sm_hdf5_netcdf %}
     hdf5:
       externals:
-      - spec: hdf5@1.12.2.3~cxx+fortran+hl~java+mpi+shared
-        prefix: /opt/cray/pe/hdf5-parallel/1.12.2.3/GNU/9.1
-        modules:
-        - cray-hdf5-parallel/1.12.2.3
+      - spec: hdf5@1.12.2.9~cxx+fortran+hl~java+mpi+shared
+        prefix: /opt/cray/pe/hdf5-parallel/1.12.2.9/gnu/12.
       buildable: false
     parallel-netcdf:
       externals:
-      - spec: parallel-netcdf@1.12.3.3+cxx+fortran+pic+shared
-        prefix: /opt/cray/pe/parallel-netcdf/1.12.3.3/GNU/9.1/
+      - spec: parallel-netcdf@1.12.3.9+cxx+fortran+pic+shared
+        prefix: /opt/cray/pe/parallel-netcdf/1.12.3.9/gnu/12.3
```

Comment:
For some reason, the module was giving us the path to the nvidia version, not the gnu version.  So I'm trying prefixes again.